doc: update Hubble doc for 1.9

rolinh opened the pull request:
Please, see individual commits for details.

Ref: #13311

rolinh committed:
doc: replace references to HUBBLE_DEFAULT_SOCKET_PATH by HUBBLE_SERVER

As of Hubble v0.7.0, which ships with Cilium 1.9, this environment
variable is now deprecated. Let's encourage the use of the new one
instead.

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

rolinh committed:
doc/observability: fix typos, make it clear Hubble is enabled by default

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

rolinh committed:
doc/clustermesh: remove reference to hubble in distributed mode

The documentation should be refering to Hubble Relay when talking about
cluster-wide visibility.

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

rolinh committed:
doc/observability: remove reference to local/distributed hubble modes

Let's consistently refer to Hubble Relay for cluster-wide visibility.
While here, update the instructions for the new method of automatic
TLS certificates generation (kubernetes cronjob).

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

pchaigno commented on Documentation/gettingstarted/hubble-relay-enable.rst:
```diff
@@ -0,0 +1,88 @@
+
+.. code:: bash
+
+   kubectl rollout restart -n $CILIUM_NAMESPACE ds/cilium
```
Not sure I understand this. Are there cases where `helm upgrade` may not restart the Cilium pods? Isn't the validation step below enough to catch corner cases?

rolinh replied:
These instructions are not new. However, when testing, I noticed that the `helm upgrade` command automatically restarts the Cilium pods but thought that the rollout restart instructions were there for a reason. However, as you point out, it might not be necessary thus I have now removed it.

pchaigno commented on Documentation/gettingstarted/hubble-relay-enable.rst:
```diff
@@ -0,0 +1,88 @@
+----------------
+
+Hubble UI relies on Hubble Relay to provide a graphical UI giving a view on the cluster.
+To enabled Hubble UI, use the following Helm upgrade command:
```
```suggestion
To enable Hubble UI, use the following Helm upgrade command:
```

pchaigno reviewed (commented):
Couple nits below. I reviewed the docs structure more than its technical content.

rolinh committed:
doc: simplify Hubble installation instructions

Now that Hubble is enabled by default, Hubble installation instructions
can be simplified to only instruct about enabling Hubble Relay for
cluster-wide visibility. Details about the Hubble architecture are now
omitted to ensure the instructions are easy to follow. A new link that
points to the observability section of the documentation is added. In
this section, users will find details about Hubble architecture should
they be interested.

Co-authored-by: Sebastian Wicki <sebastian@isovalent.com>
Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>
Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>

michi-covalent commented on Documentation/gettingstarted/clustermesh.rst:
```diff
@@ -423,15 +423,14 @@ the ``--cluster-name`` agent option or ``cluster-name`` ConfigMap option.
+Hubble Relay need to be signed by the same Certificate Authority (CA) for both
+clusters. This can be achieved by disabling automatic TLS certificate
+generation and manually providing certificates as instructed in
+:ref:`hubble_configure_tls_certs`.
```
ah i see i was hoping this would magically work with auto tls generation 😉

michi-covalent reviewed (approved):
lgtm 💯  thanks @Rolinh i'll try following the instructions as well.

michi-covalent commented on Documentation/gettingstarted/hubble-enable.rst:
```diff
@@ -1,96 +1,67 @@
-Enable Hubble
-==============
+Enable Hubble for Cluster-Wide Visibility
```
in some sense hubble is already enabled. how about:
```suggestion
Enable Hubble Relay for Cluster-Wide Visibility
```

gandro replied:
I think Robin did this after discussing the doc changes with me. The title intentionally does not mention Hubble Relay, therefore let me defend this change :) 

From the perspective of someone who is trying out Cilium and Hubble for the first time, they should not worry about what components actually make Hubble work. We want users to associate "using Hubble" with "using Hubble Relay", as it is becoming the main entry point for the Hubble API and also a requirement for being able to use the UI (which I think is a much better first introduction to Hubble than the CLI).

While local mode is enabled by default, it's hard to use and does not come with the UI. While we don't need to hide this fact, it's probably not something we want people to worry about when using Hubble for the first time. So while "Enabling Hubble Relay" is absolutely the technically more correct here, from a users' perspective I think "Enabling Hubble" sends the better message.

Advanced users will have to read up on the more in-depth concepts section anyway to learn about how Hubble works internally and how to configure the different components individually.

michi-covalent replied:
well defended 👍

michi-covalent commented on Documentation/gettingstarted/hubble-enable.rst:
```diff
@@ -1,96 +1,67 @@
-for cloud native workloads. It is built on top of Cilium and eBPF to enable
-deep visibility into the communication and behavior of services as well as the
-networking infrastructure in a completely transparent manner.
+Hubble is the component for observability in Cilium. Cluster-wide visibility is
```
while following the instructions, this paragraph felt a bit out of place. it might make it a bit smoother if we move this to the bottom and simply say:
```
.. note:

   See the :ref:`concepts_observability` section for more information about observability in Cilium.
```
